### ISSUE TITLE
Start Frescobaldi with a file or a session crash Frescobaldi

### ISSUE THREAD

**Issue by tobik** [wbsoft/frescobaldi#694]
When I open Frescobaldi by clicking a .ly file or with a active, none empty session Frescobaldi crash reproducable. There is only the Windows message box: "frescobaldi.exe does not work anymore".

This is a with both Frescobaldi 2.18.0 and 2.18.1 on Windows 8.1.

To recover I start Frescobaldi from the command line with /? argument. This start Frescobaldi but don't open a document. 

What can I do to provide you more information?

**Comment by tobik** [comment]
This is only if there is the pdf file existing. When I delete the pdf file, I can open the Frescobaldi by clicking the ly file without crash.

**Comment by PeterBjuhr** [comment]
Could it be an error in the path when the existing pdf is to be opened?

**Comment by tobik** [comment]
Well, the PDF file get opened automatically when the ly file is clicked. So the path should be correct, because the same ly file opens, when the PDF got removed. 

I also tried local path, network path and path name without whitespace, but it is all the same. 

<div>-------- Ursprüngliche Nachricht --------</div><div>Von: Peter Bjuhr <notifications@github.com> </div><div>Datum:31.05.2015  10:17  (GMT+01:00) </div><div>An: wbsoft/frescobaldi <frescobaldi@noreply.github.com> </div><div>Cc: tobik <tobias.kretschmar@gmx.de> </div><div>Betreff: Re: [frescobaldi] Start Frescobaldi with a file or a session crash
  Frescobaldi (#694) </div><div>
</div>Could it be an error in the path when the existing pdf is to be opened?

**Comment by PeterBjuhr** [comment]
When you delete the pdf and open the program, I understand that Frescobaldi doesn't crash when you recreate the pdf (by engraving)!??

From your description I also take it that this doesn't just happen to a specific file but is a more general problem!?

**Comment by tobik** [comment]
That's  correct

<div>-------- Ursprüngliche Nachricht --------</div><div>Von: Peter Bjuhr <notifications@github.com> </div><div>Datum:31.05.2015  11:12  (GMT+01:00) </div><div>An: wbsoft/frescobaldi <frescobaldi@noreply.github.com> </div><div>Cc: tobik <tobias.kretschmar@gmx.de> </div><div>Betreff: Re: [frescobaldi] Start Frescobaldi with a file or a session crash
  Frescobaldi (#694) </div><div>
</div>When you delete the pdf and open the program, I understand that Frescobaldi doesn't crash when you recreate the pdf (by engraving)!??

From your description I also take it that this doesn't just happen to a specific file but is a more general problem!?

**Comment by PeterBjuhr** [comment]
Thanks, I'll try to reproduce this on Windows later.

**Comment by tobik** [comment]
BTW: The error occurs only on my new Thinkpad W550s. On the old T61 it
works ok. IIRC in the beginning it worked ok on the W550s too.



Am 31.05.2015 um 11:43 schrieb Peter Bjuhr:

**Comment by tobik** [comment]
I just removed the entire frescobaldi tree from the windows registry and now it is working again. I have a backup of the tree available and will dig into it tonight. Please standby.

**Comment by tobik** [comment]
When I remove this item from the original registry, the crashes on startup by clicking a ly file do vanish:

[HKEY_CURRENT_USER\Software\frescobaldi\frescobaldi\mainwindow]
"state"=hex:40,00,42,00,79,00,74,00,65,00,41,00,72,00,72,00,61,00,79,00,28,00,\
  00,00,00,00,00,00,ff,00,00,00,00,00,00,00,00,00,fd,00,00,00,00,00,00,00,04,\
  00,00,00,00,00,00,00,00,00,00,00,00,00,01,00,4f,00,00,00,00,00,05,00,ae,00,\
  fc,00,02,00,00,00,00,00,00,00,06,00,fb,00,00,00,00,00,00,00,20,00,00,00,71,\
  00,00,00,75,00,00,00,69,00,00,00,63,00,00,00,6b,00,00,00,69,00,00,00,6e,00,\
  00,00,73,00,00,00,65,00,00,00,72,00,00,00,74,00,00,00,70,00,00,00,61,00,00,\
  00,6e,00,00,00,65,00,00,00,6c,00,00,00,00,00,00,00,00,00,4c,00,00,00,00,00,\
  05,00,ae,00,00,00,00,00,01,00,41,00,00,00,ff,00,ff,00,ff,00,fb,00,00,00,00,\
  00,00,00,0e,00,00,00,63,00,00,00,68,00,00,00,61,00,00,00,72,00,00,00,6d,00,\
  00,00,61,00,00,00,70,00,00,00,00,00,00,00,00,00,00,00,ff,00,ff,00,ff,00,ff,\
  00,00,00,00,00,00,00,1b,00,00,00,ff,00,ff,00,ff,00,fb,00,00,00,00,00,00,00,\
  18,00,00,00,64,00,00,00,6f,00,00,00,63,00,00,00,75,00,00,00,6d,00,00,00,65,\
  00,00,00,6e,00,00,00,74,00,00,00,6c,00,00,00,69,00,00,00,73,00,00,00,74,00,\
  00,00,00,00,00,00,00,00,00,00,ff,00,ff,00,ff,00,ff,00,00,00,00,00,00,00,1b,\
  00,00,00,ff,00,ff,00,ff,00,fb,00,00,00,00,00,00,00,18,00,00,00,6f,00,00,00,\
  75,00,00,00,74,00,00,00,6c,00,00,00,69,00,00,00,6e,00,00,00,65,00,00,00,70,\
  00,00,00,61,00,00,00,6e,00,00,00,65,00,00,00,6c,00,00,00,00,00,00,00,00,00,\
  00,00,ff,00,ff,00,ff,00,ff,00,00,00,00,00,00,00,1b,00,00,00,ff,00,ff,00,ff,\
  00,fb,00,00,00,00,00,00,00,28,00,00,00,6c,00,00,00,61,00,00,00,79,00,00,00,\
  6f,00,00,00,75,00,00,00,74,00,00,00,63,00,00,00,6f,00,00,00,6e,00,00,00,74,\
  00,00,00,72,00,00,00,6f,00,00,00,6c,00,00,00,6f,00,00,00,70,00,00,00,74,00,\
  00,00,69,00,00,00,6f,00,00,00,6e,00,00,00,73,00,00,00,00,00,00,00,00,00,4c,\
  00,00,00,00,00,05,00,73,00,00,00,00,00,00,00,1b,00,00,00,ff,00,ff,00,ff,00,\
  fb,00,00,00,00,00,00,00,18,00,00,00,6f,00,00,00,62,00,00,00,6a,00,00,00,65,\
  00,00,00,63,00,00,00,74,00,00,00,65,00,00,00,64,00,00,00,69,00,00,00,74,00,\
  00,00,6f,00,00,00,72,00,00,00,00,00,00,00,00,00,00,00,ff,00,ff,00,ff,00,ff,\
  00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,01,00,\
  00,00,00,00,04,00,55,00,00,00,00,00,05,00,78,00,fc,00,02,00,00,00,00,00,00,\
  00,01,00,fc,00,00,00,00,00,00,00,4c,00,00,00,00,00,05,00,78,00,00,00,00,00,\
  00,00,7a,00,00,00,ff,00,ff,00,ff,00,fa,00,00,00,00,00,00,00,00,00,02,00,00,\
  00,00,00,00,00,03,00,fb,00,00,00,00,00,00,00,1c,00,00,00,6d,00,00,00,75,00,\
  00,00,73,00,00,00,69,00,00,00,63,00,00,00,76,00,00,00,69,00,00,00,65,00,00,\
  00,77,00,00,00,70,00,00,00,61,00,00,00,6e,00,00,00,65,00,00,00,6c,00,01,00,\
  00,00,00,00,00,00,00,00,ff,00,ff,00,ff,00,ff,00,00,00,00,00,00,00,7a,00,00,\
  00,ff,00,ff,00,ff,00,fb,00,00,00,00,00,00,00,16,00,00,00,68,00,00,00,65,00,\
  00,00,6c,00,00,00,70,00,00,00,62,00,00,00,72,00,00,00,6f,00,00,00,77,00,00,\
  00,73,00,00,00,65,00,00,00,72,00,00,00,00,00,00,00,00,00,00,00,ff,00,ff,00,\
  ff,00,ff,00,00,00,00,00,00,00,1b,00,00,00,ff,00,ff,00,ff,00,fb,00,00,00,00,\
  00,00,00,18,00,00,00,73,00,00,00,76,00,00,00,67,00,00,00,76,00,00,00,69,00,\
  00,00,65,00,00,00,77,00,00,00,70,00,00,00,61,00,00,00,6e,00,00,00,65,00,00,\
  00,6c,00,00,00,00,00,00,00,00,00,00,00,ff,00,ff,00,ff,00,ff,00,00,00,00,00,\
  00,00,1b,00,00,00,ff,00,ff,00,ff,00,00,00,00,00,00,00,02,00,00,00,00,00,04,\
  00,8c,00,00,00,00,00,00,00,db,00,fc,00,01,00,00,00,00,00,00,00,01,00,fb,00,\
  00,00,00,00,00,00,10,00,00,00,6d,00,00,00,69,00,00,00,64,00,00,00,69,00,00,\
  00,74,00,00,00,6f,00,00,00,6f,00,00,00,6c,00,01,00,00,00,00,00,00,00,00,00,\
  00,00,00,00,04,00,8c,00,00,00,00,00,00,00,80,00,00,00,ff,00,ff,00,ff,00,00,\
  00,00,00,00,00,03,00,00,00,00,00,04,00,8c,00,00,00,00,00,01,00,43,00,fc,00,\
  01,00,00,00,00,00,00,00,03,00,fb,00,00,00,00,00,00,00,0e,00,00,00,6c,00,00,\
  00,6f,00,00,00,67,00,00,00,74,00,00,00,6f,00,00,00,6f,00,00,00,6c,00,01,00,\
  00,00,00,00,00,00,00,00,00,00,00,00,04,00,8c,00,00,00,00,00,00,00,66,00,00,\
  00,ff,00,ff,00,ff,00,fb,00,00,00,00,00,00,00,16,00,00,00,73,00,00,00,6e,00,\
  00,00,69,00,00,00,70,00,00,00,70,00,00,00,65,00,00,00,74,00,00,00,74,00,00,\
  00,6f,00,00,00,6f,00,00,00,6c,00,00,00,00,00,00,00,00,00,00,00,ff,00,ff,00,\
  ff,00,ff,00,00,00,00,00,00,00,66,00,00,00,ff,00,ff,00,ff,00,fb,00,00,00,00,\
  00,00,00,1a,00,00,00,6d,00,00,00,69,00,00,00,64,00,00,00,69,00,00,00,69,00,\
  00,00,6e,00,00,00,70,00,00,00,75,00,00,00,74,00,00,00,74,00,00,00,6f,00,00,\
  00,6f,00,00,00,6c,00,00,00,00,00,00,00,00,00,00,00,ff,00,ff,00,ff,00,ff,00,\
  00,00,00,00,00,00,66,00,00,00,ff,00,ff,00,ff,00,00,00,00,00,04,00,8c,00,00,\
  00,00,00,03,00,48,00,00,00,00,00,00,00,01,00,00,00,00,00,00,00,02,00,00,00,\
  00,00,00,00,01,00,00,00,00,00,00,00,02,00,fc,00,00,00,00,00,00,00,01,00,00,\
  00,00,00,00,00,02,00,00,00,00,00,00,00,02,00,00,00,00,00,00,00,18,00,00,00,\
  74,00,00,00,6f,00,00,00,6f,00,00,00,6c,00,00,00,62,00,00,00,61,00,00,00,72,\
  00,00,00,5f,00,00,00,6d,00,00,00,61,00,00,00,69,00,00,00,6e,00,01,00,00,00,\
  00,00,00,00,00,00,ff,00,ff,00,ff,00,ff,00,00,00,00,00,00,00,00,00,00,00,00,\
  00,00,00,00,00,00,00,00,00,00,00,1a,00,00,00,74,00,00,00,6f,00,00,00,6f,00,\
  00,00,6c,00,00,00,62,00,00,00,61,00,00,00,72,00,00,00,5f,00,00,00,6d,00,00,\
  00,75,00,00,00,73,00,00,00,69,00,00,00,63,00,01,00,00,00,00,00,01,00,dc,00,\
  ff,00,ff,00,ff,00,ff,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,00,29,\
  00

**Comment by wbsoft** [comment]
thanks for all the research!
What happens when you remove or comment line 353 in mainwindow.py?

```python
     # self.restoreState(settings.value('state', QByteArray(), QByteArray))
```

**Comment by tobik** [comment]
Yes, that fixes it too.

**Comment by tobik** [comment]
Another observation - when I close the MIDI-Player Frescobaldi starts without crash. Deleting the midi file does not prevent the crash.

**Comment by tobik** [comment]
If I put a QMessageBox in main.py line 267 right before win.setCurrentDocument(doc) there is no crash anymore. Does the win need more time to initialize before a document can be set? Is this a QT error?

**Comment by tobik** [comment]
If I put a QMessageBox in viewmanager.py line 255 right before self.viewChanged.emit(self.activeViewSpace().activeView())  there is no crash anymore.

Does something need more time to initialize?

**Comment by tobik** [comment]
Fixed with 2.18.2
